daemon/api.go

```diff
@@ -585,21 +585,50 @@ func deleteOp(c *Command, r *http.Request) Response {
 	}
 }
 
+type licT struct {
+	Intro   string
+	License string
+	Agreed  bool
+}
+
+func (*licT) Error() string {
+	// licT isn't an error, strictly speaking, but can
+	// be converted into one for convenience
+	return "license agreement required"
+}
+
 type packageInstruction struct {
+	progress.NullProgress
```

Comment:
Why is a packageInstruction being turned into a progress meter just so it can silence operations?  Wouldn't this be cleaner as something like

    var nullProg progress.NullProgress
    ...
    snappy.Install(..., nullProg)